### DISCUSSION TITLE
nodeViewProps not reactive (Vue composition API)?

### DISCUSSION THREAD

**Discussion by jloganolson** [ueberdosis/tiptap#1728] [General Questions] [answered]
The following lines generate a warning with Vue:
```
  props: nodeViewProps,
  setup(props, context) {
    const { editor, getPos, node, deleteNode } = toRefs(props);
```

The warning:
> toRefs() expects a reactive object but received a plain one.

Any idea what's going on?

**Comment by philippkuehn** [comment]
are you using vue 3 or vue 2 + composition API?

**Reply by jloganolson** [reply]
latest vue (3.2) that just dropped this week (which I think also introduced another warning into vue nodeview)s:

> [Vue warn]: injected property "decorationClasses" is a ref and will be auto-unwrapped and no longer needs `.value` in the next minor release. To opt-in to the new behavior now, set `app.config.unwrapInjectedRef = true` (this config is temporary and will not be needed in the future.)

**Reply by Nextra** [reply]
Whatever tiptap does here seems to be incompatible with the Vue 3 composition API. 

The `props` value received in `setup` (through [`VueNodeView`](https://github.com/ueberdosis/tiptap/blob/b37309005d4f8028d27c4fd3e8cbd40e3d2abc00/packages/vue-3/src/VueNodeViewRenderer.ts#L88-L108)) is a plain object that does not react to any changes made in the editor. Whatever [`VueRenderer`](https://github.com/ueberdosis/tiptap/blob/b37309005d4f8028d27c4fd3e8cbd40e3d2abc00/packages/vue-3/src/VueRenderer.ts#L29) does to get the props to be reactive gets lost along the way to the composition-style `setup` function. 

This means none of vue reactivity toolset does anything after initial run, such as `computed` or `watch` and `watchEffect`.

Writing a component with the classic options API works as expected.

**Comment by philippkuehn** [comment, marked as answer]
ah, found the issue (you are right @Nextra)! should be fixed with [this patch](https://github.com/ueberdosis/tiptap/commit/7f7f93dc1b4b56c7647de27a792e000b3513c226).

**Reply by Nextra** [reply]
Great! Seems to work as expected now.
